symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php

```diff
@@ -406,4 +408,95 @@ public function getDefaultTimesheet(
         }
         return $qb->getQuery()->getOneOrNullResult();
     }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return TimesheetItem[]
+     */
+    public function getTimesheetItemsForEmployeeReport(
+        EmployeeReportsSearchFilterParams $filterParams
+    ): array {
+        return $this->getTimesheetItemsPaginatorForEmployeeReport($filterParams)->getQuery()->execute();
+    }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return int
+     */
+    public function getTimesheetItemsCountForEmployeeReport(EmployeeReportsSearchFilterParams $filterParams): int
+    {
+        return $this->getTimesheetItemsPaginatorForEmployeeReport($filterParams)->count();
+    }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return Paginator
+     */
+    private function getTimesheetItemsPaginatorForEmployeeReport(
+        EmployeeReportsSearchFilterParams $filterParams
+    ): Paginator {
+        $qb = $this->getTimesheetItemsForEmployeeReportQueryBuilderWrapper($filterParams)->getQueryBuilder();
+        $qb->addSelect('COALESCE(SUM(timesheetItem.duration),0) AS totalDurationByGroup');
+        $qb->addGroupBy('timesheetItem.project');
+        $qb->addGroupBy('timesheetItem.projectActivity');
+        return $this->getPaginator($qb);
+    }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return QueryBuilderWrapper
+     */
+    private function getTimesheetItemsForEmployeeReportQueryBuilderWrapper(
+        EmployeeReportsSearchFilterParams $filterParams
+    ): QueryBuilderWrapper {
+        $q = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $q->leftJoin('timesheetItem.timesheet', 'timesheet');
+        $q->leftJoin('timesheetItem.project', 'project');
+        $q->leftJoin('project.customer', 'customer');
+        $q->andWhere('timesheetItem.employee = :empNumber');
+        $q->setParameter('empNumber', $filterParams->getEmpNumber());
+        $this->setSortingAndPaginationParams($q, $filterParams);
+
+        if (!is_null($filterParams->getProjectId())) {
+            $q->andWhere('timesheetItem.project = :projectId');
+            $q->setParameter('projectId', $filterParams->getProjectId());
+        }
+
+        if (!is_null($filterParams->getActivityId())) {
+            $q->andWhere('timesheetItem.projectActivity = :activityId');
+            $q->setParameter('activityId', $filterParams->getActivityId());
+        }
+
+        //Timesheet items after fromDate (including fromDate)
+        if (!is_null($filterParams->getFromDate())) {
+            $q->andWhere($q->expr()->gte('timesheetItem.date', ':fromDate'));
+            $q->setParameter('fromDate', $filterParams->getFromDate());
+        }
+
```

Comment:
we can refactor the code snippets related to the date range filter as if elseif statement
https://github.com/orangehrm/orangehrm/pull/1086/files#diff-eba82252347cc542b14f7b1251656b28681e538802c1f2cbf7d0e1ca70201879R286